File: apiRest/packages/domain/src/mappers/gameMapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GameMapper = void 0;
const gameModel_1 = require("../models/gameModel");
class GameMapper {
    static toDynamo(game) {
        const data = game.toPrimitives();
        return {
            gameId: data.gameId,
            status: data.status,
            config: data.config,
            players: data.players,
            board: data.board,
            turn: data.turn,
            winner: data.winner,
            createdAt: data.createdAt,
            updatedAt: data.updatedAt,
            ttl: data.ttl,
        };
    }
    static fromDynamo(item) {
        return gameModel_1.GameModel.fromPrimitives({
            gameId: item.gameId,
            status: item.status,
            config: item.config,
            players: item.players,
            board: item.board,
            turn: item.turn,
            winner: item.winner,
            createdAt: item.createdAt,
            updatedAt: item.updatedAt,
            ttl: item.ttl
        });
    }
}
exports.GameMapper = GameMapper;

File: webSocket/default/mappers/game_mapper.py
# mappers/game_mapper.py
from models.game import GameModel

class GameMapper:
    @staticmethod
    def to_dynamo(game: GameModel) -> dict:
        data = game.to_primitives()

        return {
            "roomId": data["room_id"],
            "board": data["board"],
            "turn": data["turn"],
            "status": data["status"],
            "createdAt": data["created_at"],
            "ttl": data["ttl"],
        }

    @staticmethod
    def from_dynamo(item: dict) -> GameModel:
        return GameModel(
            room_id=item["roomId"],
            board=item["board"],
            turn=item["turn"],
            status=item["status"],
            created_at=item["createdAt"],
            ttl=item["ttl"],
        )
